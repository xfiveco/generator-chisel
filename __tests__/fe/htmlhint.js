const fs = require('fs-extra');
const { defaultAnswers } = require('./helpers');

describe('FE HTMLHint', () => {
  test('HTMLHint errors are logged and fail build ', async () => {
    await global.chiselTestHelpers.generateProjectWithAnswers(
      ['create'],
      defaultAnswers,
    );

    fs.outputFileSync(
      './src/templates/page-1.twig',
      /* HTML */ `
        <!DOCTYPE html>
        <div notOkAttribute>Hello</div>
      `,
    );
    fs.outputFileSync(
      './src/templates/page-2.twig',
      /* HTML */ `
        <!DOCTYPE html>
        <div>OK!</div>
      `,
    );

    fs.outputFileSync(
      './src/templates/page-3.twig',
      /* HTML */ `
        <div>
          <ul>
            <li id="li1">Line item 1</li>
            <li id="li1">Line item 2</li>
          </ul>
        </div>
      `,
    );

    const consoleMock = jest.spyOn(console, 'log');

    const build = global.chiselTestHelpers.runChiselScript(['build']);

    await expect(build).rejects.toThrowErrorMatchingSnapshot();

    global.chiselTestHelpers.normalizeConsoleMockCalls(consoleMock);
    expect(consoleMock.mock.calls).toMatchSnapshot();
  });
});
