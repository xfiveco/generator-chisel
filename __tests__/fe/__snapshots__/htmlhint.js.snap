// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FE HTMLHint HTMLHint errors are logged and fail build  1`] = `"Problems found when checking with HTML Hint"`;

exports[`FE HTMLHint HTMLHint errors are logged and fail build  2`] = `
Array [
  Array [],
  Array [
    "[1m--PROJECT-PATH--/dist/page-1.html:[22m
[37mL2 |[90m<div notOkAttribute>Hello</div>[39m
[37m        ^ [33mThe attribute name of [ notOkAttribute ] must be in lowercase. (attr-lowercase)[39m

[1m--PROJECT-PATH--/dist/page-3.html:[22m
[37mL1 |[90m<div>[39m
[37m    ^ [33mDoctype must be declared first. (doctype-first)[39m
[37mL4 |[90m    <li id=\\"li1\\">Line item 2</li>[39m
[37m           ^ [33mThe id value [ li1 ] must be unique. (id-unique)[39m",
  ],
  Array [],
]
`;
